app/src/main/java/com/sdp13epfl2021/projmag/database/UserDataDatabase.kt

```diff
@@ -0,0 +1,68 @@
+package com.sdp13epfl2021.projmag.database
+
+import java.lang.Exception
+
+/**
+ * An interface to manage a database containing
+ * user data
+ *
+ */
+interface UserDataDatabase {
+    /**
+     * Push the ID of a favorite `Project` to the database
+     * Call `onSuccess` if the operation succeeded
+     * Call `onFailure` with an Exception in case of failure
+     *
+     * @param projectID The ID of the favorite project
+     * @param onSuccess called on success
+     * @param onFailure called with an exception on failure
+     */
+    fun pushFavoriteProject(
+        projectID: ProjectId,
+        onSuccess: () -> Unit,
+        onFailure: (Exception) -> Unit
```

Comment:
Since each user (or just student), will have different favorites, this interface seems to be not correct for our case.
Maybe after defining a User and/or Student, we could have something like : pushFavoriteProject(projectId, userId, ....)
Another possibility is to only keep informations of favorites project, in local.